subsys/net/ip/l2/ethernet.c

```diff
@@ -83,7 +83,7 @@ static inline void ethernet_update_length(struct net_if *iface,
 			NET_IPV6H_LEN;
 	}
 
-	if (len < NET_ETH_MINIMAL_FRAME_SIZE - sizeof(struct net_eth_hdr)) {
+	if (len < net_buf_frags_len(pkt->frags)) {
```

Comment:
This changes the semantics of this check. Instead of validating the ethernet frame, you are checking the IP packet length and adjusting the whole packet. There are separate checks in IPv6 and IPv4 modules that discard the packet if the length does not match the actual data received. That check would not work after this. Also I would like to avoid calculating the packet length, meaning calling net_buf_frags_len() for each received ethernet frame as that is somewhat expensive.
It also seems that for USB ethernet, the minimum ethernet packet length is not 65 bytes or is it? If USB ethernet can send shorter packets, then we would probably need to do this check lower in the stack meaning in the USB driver.

Reply:
Thanks for the feedback!  The issue is when the packet is a multiple of 64 bytes long as the Linux usbnet driver automatically adds a single extra zero byte to terminate the packet.

I'll change the code in the USB driver to do the discard instead.